### ISSUE TITLE
Render during reconciliation causes NPE

### ISSUE THREAD

**Issue by sambev** [facebook/react#6027]
I have a view with a controlled select on it. When the onChange event fires, the resulting function actually changes the view, removing the select. Then I get this error:

`Uncaught TypeError: Cannot set property 'pendingUpdate' of null`

Reproduced here:
https://jsfiddle.net/wntgpwcx/5/  (react 0.14.7)

I also noticed I don't get the same error in an older version (0.13.1)
https://jsfiddle.net/wntgpwcx/6/

The line that throws is in ReactDOMSelect L183:
```javascript
function _handleChange(event) {
  var props = this._currentElement.props;
  // _wrapper state is not null here
  var returnValue = LinkedValueUtils.executeOnChange(props, event);

  this._wrapperState.pendingUpdate = true; // _wrapperState is null after the executeOnChange happens
  ReactUpdates.asap(updateOptionsIfPendingUpdateAndMounted, this);
  return returnValue;
}
```

I was able to see a fix for a similar situation here:
https://github.com/facebook/react/pull/4624/files

I don't know if a similar check could be applied here, or maybe I just shouldn't remove my select as a result of an onChange event?

**Comment by jimfb** [comment]
Probably affects unstable_subtree also.

I don't think we need a fix this for the v15 milestone since it was a regression introduced in 0.14 and this is the first we're hearing about it, but I think it is a valid bug.

**Comment by sambev** [comment]
@jimfb Would the check in the PR I linked be a sufficient check to add? I would be more than happy to add it in myself.

**Comment by jimfb** [comment]
@sambev Sorry, I'm not sure I understand your question.  The PR you linked was merged.

**Comment by sambev** [comment]
@jimfb Sorry, I wasn't that clear. I simply applied the same conditional in that PR, in https://github.com/facebook/react/pull/6028. However, I don't know the full impact of that change.

**Comment by jimfb** [comment]
Fixed in #6028 and verified in http://jsfiddle.net/uptxgjvd/
